src/imp/openssl.rs

```diff
@@ -270,6 +270,16 @@ impl TlsConnector {
             }
         }
 
+        if builder.disable_built_in_roots {
+            let mut verify_cert_store = X509StoreBuilder::new()?;
+            for cert in &builder.root_certificates {
+                if let Err(err) = verify_cert_store.add_cert((cert.0).0.clone()) {
+                    debug!("add_cert error: {:?}", err);
+                }
+            }
+            connector.set_verify_cert_store(verify_cert_store.build())?;
+        }
```

Comment:
This should probably go before the `add_cert`s so it doesn't have to re-add the `root_certificates`?

Reply:
Never mind, apparently `cert_store_mut` accesses a different cert store...

Reply:
Dug into this a bit, I'll summarize the findings here.

Besides the `ctx->cert_store`, OpenSSL has two other optional cert stores, the `cert->verify_store` and the `cert->chain_store`. If the `verify_store` is set, OpenSSL will use it instead of `ctx->cert_store` in [`ssl_verify_cert_chain`](https://github.com/openssl/openssl/blob/e15d369781eb2e97656ebbabef576b8079d86b8c/ssl/ssl_cert.c#L384). If the `chain_store` is set, OpenSSL will use it instead of `ctx->cert_store` in [`ssl_build_cert_chain`](https://github.com/openssl/openssl/blob/e15d369781eb2e97656ebbabef576b8079d86b8c/ssl/ssl_cert.c#L861) and [`ssl_add_cert_chain`](https://github.com/openssl/openssl/blob/beb958ccd87b95f1a86bfe2b879492b62e58d80e/ssl/statem/statem_lib.c#L931).

It does look like we can skip the `connector.cert_store_mut().add_cert((cert.0).0.clone())` above if we use `set_verify_cert_store` and my tests will still pass. Haven't dug far enough into OpenSSL to know why that is.

Reply:
Just to avoid running into any edge cases, let's use SSL_CTX_set_cert_store instead to keep things consistent. I'll add that to rust-openssl and cut a release for you.

Reply:
If we're going to change rust-openssl I think it'd be preferable to opt out of the `set_default_verify_paths` call.

Reply:
That would be a more invasive API change since it currently happens directly in the constructor of the SslConnectorBuilder.

Reply:
True. I suppose it's not a big deal.

Reply:
openssl 0.10.29 has the new method.

Reply:
Thanks! I'll update shortly.